Display Circuit members

peterschwarz opened the pull request:
This PR implements the member list for circuit details.
![Screen Shot 2020-06-17 at 11 32 46 AM](https://user-images.githubusercontent.com/282233/84925423-7a7c4d80-b08f-11ea-9090-90a394a62636.png)

A future PR will add the service expanders for each row.

peterschwarz committed:
Add API calls for proposal and circuit

Add calls to the splinter API for proposals and circuits based on
circuit id.

Signed-off-by: Peter Schwarz <pschwarz@bitwise.io>

peterschwarz committed:
Add circuitState for single circuit

Add a React state with effects for a single circuit.  This state will
fetch the circuit by id.  If it doesn't exist, it will try the proposal
for the same.

This state is triggered off of the circuit id: the user can't set the
state of this server-based value.

Signed-off-by: Peter Schwarz <pschwarz@bitwise.io>

peterschwarz committed:
Link to circuit details

Link the circuit table rows to the circuit details page.

Signed-off-by: Peter Schwarz <pschwarz@bitwise.io>

peterschwarz committed:
Correct actionRequired to account for requester

Account for the node being the requester when indicating if action is
required on a circuit by a particular node.

Signed-off-by: Peter Schwarz <pschwarz@bitwise.io>

eloaverona commented on saplings/circuits/src/components/circuitDetails/CircuitDetails.js:
```diff
@@ -0,0 +1,165 @@
+  const [fullNodes, setNodes] = React.useState(null);
+
+  React.useEffect(() => {
+    const fetchNodes = async () => {
```
You could use the `useNodeRegistryState` in `state/nodeRegistry` instead of repeating the request here. Is there a reason you did not use it?

peterschwarz replied:
The effect, in this case, is based off the change of `nodes`, which is actually a collection of node ID values.  They operate differently. I could modify the  node registry state to be a reducer, but that seemed a bit overly complex.

eloaverona commented on saplings/circuits/src/components/circuitDetails/CircuitDetails.js:
```diff
@@ -0,0 +1,165 @@
+          <td>
+            {node.metadata.company || node.metadata.organization || 'N/A'}
+          </td>
+          <td>{node.endpoints[0] || 'N/A'}</td>
```
Is there are reason to display only one endpoint?

peterschwarz replied:
Other than there being no example with multiples, no.  I could list them out, each on a new line.

eloaverona commented on saplings/circuits/src/components/circuitDetails/CircuitDetails.js:
```diff
@@ -0,0 +1,165 @@
+    <div className="table-container">
+      <table className="nodes-table">
+        <tr className="table-header">
+          <th>Name</th>
```
This should be ID

peterschwarz replied:
Fixed

eloaverona commented on saplings/circuits/src/components/circuitDetails/CircuitDetails.js:
```diff
@@ -0,0 +1,165 @@
+
+const NodeStatus = ({ circuit, nodeId }) => {
+  if (circuit.awaitingApproval() && !circuit.nodeHasVoted(nodeId)) {
+    return <span className="status awaiting-approval">Awaiting approval</span>;
```
The rules for the classes `status` and `awaiting-approval` don't seen to be being applied here

peterschwarz replied:
I will probably deal with those when working on the status tags.

eloaverona commented on saplings/circuits/src/components/circuitDetails/CircuitDetails.scss:
```diff
@@ -0,0 +1,94 @@
+  }
+}
+
+.service-container {
```
Should all these rules about "service-container" be included in this commit?

peterschwarz replied:
Removed

eloaverona commented on saplings/circuits/src/components/MainHeader.scss:
```diff
@@ -48,4 +48,22 @@
+      color: var(--color-grey);
+      padding: 0.25rem;
+    }
+  }
```
Rules for the class `managementType` are already set in the commit `Add circuit details page` f267a3f76f52e0aafd72980a3cc832106c5752fb . They are duplicated here

peterschwarz replied:
Yeah, seems like some mistake with my commits.  I moved these to the `CircuitDetails.scss` file, so this is no longer relevant here.

eloaverona commented on saplings/circuits/src/components/MainHeader.scss:
```diff
@@ -15,15 +15,37 @@
-  align-items: center;
   display: flex;
-  flex-direction: row;
+  flex-direction: column;
```
This change also affected the main page. Was it intentional? 
Before: 
<img width="400" alt="Screen Shot 2020-06-18 at 7 41 31 AM" src="https://user-images.githubusercontent.com/14094978/85021242-33e22e00-b137-11ea-9a8f-de30018741fe.png">

After:
<img width="400" alt="Screen Shot 2020-06-18 at 7 39 54 AM" src="https://user-images.githubusercontent.com/14094978/85021098-f9789100-b136-11ea-9c34-d2a187ff667d.png">

eloaverona commented on saplings/circuits/src/components/circuitDetails/CircuitDetails.js:
```diff
@@ -0,0 +1,165 @@
+        </div>
+        <div className="circuits-title">
+          <h4>
+            Circuit
```
There should be a space between the word circuit and the circuit ID. Right now they display as a single word:
<img width="531" alt="Screen Shot 2020-06-18 at 7 44 59 AM" src="https://user-images.githubusercontent.com/14094978/85021578-a5ba7780-b137-11ea-83c8-f5d7d9674a9c.png">

peterschwarz replied:
Fixed

eloaverona reviewed (changes requested):
It was a big request in the design discussion that a circuits and proposals can be easily differentiated in the circuit details page. It was the intention behind the big "awaiting approval" above the circuit name in the mock ups @RyanLassigBanks created. It does not need to be added in this PR, but definitely something like that should be added to this page.  
Here is the mock up for reference: 
<img width="400" alt="Screen Shot 2020-06-15 at 5 57 43 PM" src="https://user-images.githubusercontent.com/14094978/85022159-8708b080-b138-11ea-9a8a-e9f7805bf6fa.png">

eloaverona reviewed (changes requested):
I just noticed that when I click in the Propose Circuit button in the main screen, I'm redirected to `/circuits/propose` but I only see a blank screen. The form is not being loaded.

peterschwarz committed:
Add circuit details page

This commit adds an initial circuit detail page, that includes the base
information about the circuit.  More details on the circuit and its
nodes will be added in future commits.

Signed-off-by: Peter Schwarz <pschwarz@bitwise.io>

peterschwarz commented:
The navigation issue has been fixed.  

Likewise, I added the "Awaiting Approval" header, but only for circuits that require a vote by the local node.  I.e. if your node was the proposing node, it's considered a vote, so no action is require.
